grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/atom/AtomicFactory.java

```diff
@@ -38,16 +47,30 @@
  *
  */
 public class AtomicFactory {
-    
+
+    /**
+     * @param atomType to create
+     * @param var atom variable name
+     * @param predicateVar predicate variable name
+     * @param predicate atom's predicate
+     * @param parent reasoner query the atom should belong to
+     * @param <T> atom type parameter
+     * @return fresh ontological atom
+     */
+   public static <T extends OntologicalAtom> OntologicalAtom createOntologicalAtom(Class<T> atomType, Var var, Var predicateVar, @Nullable IdPredicate predicate, ReasonerQuery parent) {
+        if (atomType.equals(PlaysAtom.class)) { return PlaysAtom.create(var, predicateVar, predicate, parent); }
+        else if (atomType.equals(HasAtom.class)){ return HasAtom.create(var, predicateVar, predicate, parent); }
+        else if (atomType.equals(RelatesAtom.class)){ return RelatesAtom.create(var, predicateVar, predicate, parent); }
+        else if (atomType.equals(SubAtom.class)){ return SubAtom.create(var, predicateVar, predicate, parent); }
+        else{ throw GraqlQueryException.illegalAtomCreation(atomType.getName()); }
+    }
     /**
      * @param atom to be copied
      * @param parent query the copied atom should belong to
      * @return atom copy
      */
     public static Atomic create(Atomic atom, ReasonerQuery parent) {
-        Atomic copy = atom.copy();
-        copy.setParentQuery(parent);
-        return copy;
+        return atom.copy(parent);
```

Comment:
Maybe this `create` method can be deleted now. IntelliJ has an "inline" refactor you can use for that 👍